src/components/NotificationCenter/index.js

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { values } from "lodash";
+import { connect } from "react-redux";
+import Notification from "./Notification";
+import TodoList from "../TodoList";
+
+
+export function NotificationCenter ({notifications, dismissNotification, deactivateNotification}) {
+  return (
+    <div className="notification-center">
+      <ul>
+        {notifications.map(notification => {
+          return (
+            <li key={notification.id}>
+              <Notification
+                notification={notification}
+                onDismiss={() => dismissNotification(notification.id)}
+                onDeactivate={() => deactivateNotification(notification.id, notification.todo.id)}
+              />
+            </li>
+          )
+        })}
+      </ul>
+    </div>
+  )
+}
+
+const mapStateToProps = (state, ownProps) => {
+  return {
+    notifications: values(state.notifications).filter((notification) => notification.active)
+  }
+}
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  return {
+    dismissNotification: (id) => {
+      dispatch({
+        type:'DISMISS_NOTIFICATION',
+        payload: {
+          id
+        }
+      })
+    },
+
+    deactivateNotification: (id, todoId, todoListId, updatedKeysInTodo) => {
+      dispatch({
+        type: 'DISMISS_NOTIFICATION',
+        payload: {
+          id
+        }
+      });
+
+      dispatch({
+        type: 'UPDATE_TODO',
+        payload: {
+          todoId,
+          updatedKeysInTodo,
+          todoListId: ownProps.id,
```

Comment:
So your error comes from the fact that the id that you pass as the todoListId is wrong. Currently it is undefined.
<img width="346" alt="Capture d’écran 2020-10-29 à 09 36 35" src="https://user-images.githubusercontent.com/7590272/97544698-449a8e80-19ca-11eb-8577-fc7cbe8ac539.png">

The reason for this is that `ownProps` (the initial props of the NotificationCenter, before you connect it to redux), is an empty object (remember in you App component when creating it, you don't pass any props to it). `ownProps.id` is thus undefined. That is why it crashes with the following error:
![123244218_366971441183952_1732200495791344333_n](https://user-images.githubusercontent.com/7590272/97545248-005bbe00-19cb-11eb-8f50-14c73f828f55.png)

If we look at the code where it crashes (in reducers/todoList#L81), it makes sense:
```javascript
todoListId = action.payload.todoListId;
todoList = state[todoListId];
todos = todoList.todos;
```
If todoListId is undefined, then todoList is undefined because `state[undefined]` won't find anything. Then when trying to access the `todos` of `undefined`, it crashes. Because `undefined` is not an object and has no "property called todos" .

The good news is, you are very close to complete the part 10 ! After fixing this issue, you will see that it works as expected :)

To fix the issue, you need to use the todoListId contained in the notification : that is why we stored it to begin with. 
1°) Call your function with three arguments from your component:
`onDeactivate={() => deactivateNotification(notification.id, notification.todoListId, notification.todo.id)}`
2°) Change the [signature](https://medium.com/@ludico8/function-ludico8-arguments-aa39a4adf9d#:~:text=The%20function%20signature%20consists%20of,Javascript%2C%20arguments%20are%20passed%20positionally.) of your function to `deactivateNotification: (id, todoListId, todoId) => {...}`
3°) Pass the correct payload: 
```javascript
payload: {
    todoId,
    todoListId,
    updatedKeysInTodo: { active: false }
}
```
And it should work !